src/rotation/texture_pile.zig

```diff
@@ -73,20 +73,26 @@ pub fn onBlockBreaking(_: ?main.items.Item, _: Vec3f, _: Vec3f, currentData: *Bl
 	if(currentData.data == 0) {
 		currentData.* = .{.typ = 0, .data = 0};
 	} else {
-		currentData.data = currentData.data - 1;
+		currentData.data = @min(currentData.data, currentData.modeData() - 1) - 1;
 	}
 }
 
+fn isThisItem(block: Block, item: main.items.ItemStack) bool {
+	return item.item != null and item.item.? == .baseItem and item.item.?.baseItem.block == block.typ;
+}
+
 pub fn canBeChangedInto(oldBlock: Block, newBlock: Block, item: main.items.ItemStack, shouldDropSourceBlockOnSuccess: *bool) RotationMode.CanBeChangedInto {
 	switch(RotationMode.DefaultFunctions.canBeChangedInto(oldBlock, newBlock, item, shouldDropSourceBlockOnSuccess)) {
 		.no, .yes_costsDurability, .yes_dropsItems => return .no,
-		.yes, .yes_costsItems => {
-			const amountChange = @as(i32, newBlock.data) - if(oldBlock.typ == newBlock.typ) @as(i32, oldBlock.data) else 0;
-			if(amountChange <= 0) {
-				return .{.yes_dropsItems = @intCast(-amountChange)};
+		.yes_costsItems => |r| return .{.yes_costsItems = r},
+		.yes => {
+			const oldData = @min(oldBlock.data, oldBlock.modeData() - 1);
```

Comment:
This doesn't handle the case of replacing non-solid blocks correctly:
```suggestion
			const oldAmount = if(oldBlock.typ == newBlock.typ) @min(oldBlock.data, oldBlock.modeData() - 1) else 0;
```